pkgs/test_core/lib/src/runner.dart

```diff
@@ -167,21 +166,12 @@ class Runner {
     var testOn = _config.suiteDefaults.metadata.testOn;
     if (testOn == PlatformSelector.all) return;
 
-    var runtimes = _config.suiteDefaults.runtimes
+    var unsupportedRuntimes = _config.suiteDefaults.runtimes
```

Comment:
This is actually just a revert of what I had tried to do earlier, it should match the original logic now. What I was trying to do no longer makes sense with the fallback to a default compiler on any runtime that has no compiler specified.